samples/interop/Xamarin.Mac.Interop/AvaloniaView.cs

```diff
@@ -12,7 +13,8 @@ public class AvaloniaView : IDisposable
 
     public AvaloniaView()
     {
-        var topLevelImpl = EmbeddedTopLevelImpl.Create();
+        var topLevelImpl = AvaloniaLocator.Current
+            .GetRequiredService<IWindowingPlatform>().CreateEmbeddableTopLevel();
```

Comment:
If you change EmbeddableControlRoot default constructor to use CreateEmbeddableTopLevel instead of CreateEmbeddableWindow, you can avoid impl-related APIs here.